### PR TITLE
Initial stab at workspace-or-crate repos

### PR DESCRIPTION
**Pull request by jcape**
- Add choice variable
- Add post-gen hook to delete the "wrong" directory.

### Motivation

Sometimes (most times, honestly), we're going to want a single-crate in a respository.

### TODO

- [x] Fixup top-level README
- [x] Test

### CODE REVIEW

**Comment by jcape** [comment]
Current dependencies on/for this PR:
* main
  * **PR #15** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/cookiecutters/15" target="_blank">Graphite</a> 
    * **PR #16** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/cookiecutters/16" target="_blank">Graphite</a> 
      * **PR #17** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/cookiecutters/17" target="_blank">Graphite</a> 
        * **PR #18** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/cookiecutters/18" target="_blank">Graphite</a> 
          * **PR #21** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/cookiecutters/21" target="_blank">Graphite</a> 
            * **PR #22** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/cookiecutters/22" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/cookiecutters/22?utm_source=stack-comment).

File: rust/repo/{{ cookiecutter.repo_name }}/README.md [hunk 1]
```diff
@@ -1,17 +1,38 @@
+<!-- markdownlint-disable -->
```
Comment (nick-mobilecoin) [inline rust/repo/{{ cookiecutter.repo_name }}/README.md:1]: :x: The markdown lint disable seems suspect it's getting propagated to the rendered output.
I'm guessing that the root `.markdownlint-cli2.jsonc` needs to be updated for the new directory structure.
```json
  "ignores": [".github/pull_request_template.md", "target/", "rust-workspace/{{cookiecutter.repo_name }}/", "rust-crate/{{ cookiecutter.crate_sub_dir }}/"]   
```

**Review by nick-mobilecoin — CHANGES_REQUESTED** [review]
main blocker is the markdown lint exclusions, I think it's a cookiecutter template issue and shouldn't propagate to the rendered outputs

File: rust/repo/{{ cookiecutter.repo_name }}/README.md (hunk 1 above)
Comment (jcape) [reply rust/repo/{{ cookiecutter.repo_name }}/README.md:1]: Fixed in the crate-target PR.

**Review by nick-mobilecoin — APPROVED** [review]

**Pull request merged**